### DISCUSSION TITLE
Error on server AWS - Error: read ECONNRESET

### DISCUSSION THREAD

**Discussion by arcctg** [PrismarineJS/mineflayer#2634] [Q&A] [answered]
I created a virtual server of AWS (Amazon Web Services) on Windows platform. When I try to run a simple bot I get this error:

```
Error: read ECONNRESET
    at TCP.onStreamRead (node:internal/stream_base_commons:217:20) {
  errno: -4077,
  code: 'ECONNRESET',
  syscall: 'read'
}
```

My code:

```
const mineflayer = require('mineflayer')

const bot = mineflayer.createBot({
    host: 'example.com', 
    //port: 27022,
    version: '1.18.2',
    logErrors: true,
    username: '', 
    password: '', 
    auth: 'microsoft' 
})

    var sec = 5000;
setTimeout(function() {
console.log(bot.players)
}, sec);
```

How can I solve this problem?

**Comment by AshDoesStuf** [comment]
const bot = mineflayer.createBot({
    host: 'localhost',
    username: '', // change this to any name you like!
    port: 12345, // change this to the servers port!
})

**Reply by AshDoesStuf** [reply]
change the port to the servers port or change the host to the servers ip

**Reply by arcctg** [reply]
Thank you, but I know that there must be a nickname, port, server IP. I deleted it on purpose when I pasted the code here

**Comment by arcctg** [comment, marked as answer]
I solved this problem by creating a new virtual server
